pkgs/tools/filesystems/f3/default.nix

```diff
@@ -1,38 +1,57 @@
-{ stdenv, fetchFromGitHub
-, parted, udev
+{ stdenv, lib, fetchFromGitHub
+, parted, systemd ? null
 }:
 
 stdenv.mkDerivation rec {
   pname = "f3";
   version = "7.2";
 
-  enableParallelBuilding = true;
-
   src = fetchFromGitHub {
     owner = "AltraMayor";
     repo = pname;
     rev = "v${version}";
     sha256 = "1iwdg0r4wkgc8rynmw1qcqz62l0ldgc8lrazq33msxnk5a818jgy";
   };
 
-  buildInputs = [ parted udev ];
+  postPatch = ''
+     sed -i 's/-oroot -groot//' Makefile
+
+     for f in f3write.h2w log-f3wr; do
+      substituteInPlace $f \
+        --replace '$(dirname $0)' $out/bin
+     done
+  '';
+
+  buildInputs = [
+    parted
+  ]
+  ++ lib.optional stdenv.isLinux systemd;
```

Comment:
I took the udev dependency from https://github.com/AltraMayor/f3#install-dependencies , is it better to use udev from systemd?

Reply:
`udev` is an alias for `systemd`.

Reply:
good to know :+1: